code/game/objects/items/robot/robot_items.dm

```diff
@@ -980,3 +981,10 @@
 	if(stored)
 		. += "You are currently holding [stored]."
 		. += "<span class='notice'<i>Alt-click</i> will drop the currently stored [stored].</span>"
+
+
+/obj/item/borg/apparatus/beaker/service/Initialize()
+	. = ..()
+	stored = new /obj/item/reagent_containers/food/drinks/drinkingglass(src)
+	RegisterSignal(stored, COMSIG_ATOM_UPDATE_ICON, /atom/.proc/update_icon)
```

Comment:
Sure you have:

![image](https://user-images.githubusercontent.com/9547572/170867181-c5e6e8c6-7c04-4aad-9a3c-89cee6121245.png)

I just defined a new subtype that gets a glass instead of a beaker.

Reply:
That said, I don't need to register the signal again, because it's already registered in the parent object.